notebooks/openvino/phi-2_on_mtl.ipynb

```diff
@@ -0,0 +1,571 @@
+    "Then using the quantized model we will show how to generate code completions with the model running on Intel Meteor Lake iGPU presenting a good experience of running GenAI locally on Intel PC marking the start of the AIPC Era!\n",
+    "Then we will show how to talk with Phi-2 in a ChatBot demo running completley locally on your Laptop!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03cb49cf-bc6f-4702-a61f-227b352404cb",
+   "metadata": {},
+   "source": [
+    "## Install dependencies\n",
+    "Make sure you have the latest GPU drivers installed on your machine: https://docs.openvino.ai/2024/get-started/configurations/configurations-intel-gpu.html.\n",
+    "\n",
+    "We will start by installing the dependencies, you can either uncomment the following cell and run it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "96d8203c-34c9-41a2-95bd-3891533840a1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ! pip install optimum[openvino,nncf] torch"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5980ce40-0be1-48c1-941a-92c484d4da31",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from transformers import AutoTokenizer\n",
+    "from optimum.intel import OVModelForCausalLM, OVWeightQuantizationConfig"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48b81857-a095-43a3-8c8d-4c880b743a6e",
+   "metadata": {},
+   "source": [
+    "## Configuration\n",
+    "Here we will configure which model to load and other attributes. We will explain everything :)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "800cd7a3-a21d-4a0a-9d73-2a2d08646f99",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name = 'microsoft/phi-2'\n",
+    "save_name = './phi-2-woq4'  # Directory where the exported & quantized model will be saved\n",
+    "precision = 'f32'  # We use FP32 precision due to Phi-2 overflow issues in FP16.\n",
+    "quantization_config = OVWeightQuantizationConfig(\n",
+    "    bits=4,\n",
+    "    sym=False,  # Use asymmetric quantization\n",
+    "    group_size=128,  # Quantize weights in groups of 128 to ensure no accuracy degradation\n",
+    "    ratio=0.8,  # 80% of the model layers will be quantized to 4bit, the rest will be quantized to 8bit.\n",
+    ")\n",
+    "device = 'gpu'  # choose from ['cpu', 'gpu']\n",
+    "stateful = True  # Optimize model by setting the KV cache as part of the models state instead of as an input"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f398868-93d7-4c2d-9591-9bac8e9b701c",
+   "metadata": {},
+   "source": [
+    "With this configuration we expect the model size to reduce to around to 1.62GB:  $0.8 \\times 2.7{\\times}10^3 \\times \\frac{1}{2}\\text{B} + 0.2 * 2.7{\\times}10^3 \\times 1\\text{B} = 1.62{\\times}10^3\\text{B} = 1.62\\text{GB}$"
```

Comment:
nice addition ! could make sense to specify that the model has  2.7 billion parameters and to give a bit more context so that it's clearer for users, wdyt ?

Reply:
Added some information about the model in the first paragraph.